.github/workflows/docs-gen-and-push.yml

```diff
@@ -26,33 +26,42 @@ jobs:
     name: print relevant info
     runs-on: ubuntu-latest
     steps:
-      - run: |
-          echo "github.actor=${{ github.actor }}"
-          echo "github.action_ref=${{ github.action_ref }}"
-          echo "github.event_name=${{ github.event_name }}"
-          echo "github.head_ref=${{ github.head_ref }}"
-          echo "github.ref=${{ github.ref }}"
-          echo "github.ref_name=${{ github.ref_name }}"
-          echo "github.repository_owner=${{ github.repository_owner }}"
-          echo "github.triggering_actor=${{ github.triggering_actor }}"
+      - env:
+          GITHUB_ACTOR: ${{ github.actor }}
+          GITHUB_ACTION_REF: ${{ github.action_ref }}
+          GITHUB_EVENT_NAME: ${{ github.event_name }}
+          GITHUB_HEAD_REF: ${{ github.head_ref }}
+          GITHUB_REF: ${{ github.ref }}
+          GITHUB_REF_NAME: ${{ github.ref_name }}
+          GITHUB_REPOSITORY_OWNER: ${{ github.repository_owner }}
+          GITHUB_TRIGGERING_ACTOR: ${{ github.triggering_actor }}
```

Comment:
(A) If I understand correctly, it is not allowed to define variables whose names start with "GITHUB_", those are under the exclusive control of GitHub. If I do not understand correctly, and it is allowed, it is a bad idea anyway.

(B) I suspect that many of those are already defined. See https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables